report-generator/src/main/java/com/teamscale/report/jacoco/JaCoCoXmlReportGenerator.java

```diff
@@ -53,11 +52,36 @@ public JaCoCoXmlReportGenerator(List<File> codeDirectoriesOrArchives,
 		this.logger = logger;
 	}
 
-	/** Creates the report. */
-	public String convert(Dump dump) throws IOException {
-		ByteArrayOutputStream output = new ByteArrayOutputStream();
+
+	/**
+	 * Creates the report and writes it to a file.
+	 *
+	 * @return The file object of for the converted report or null if it could not be created
+	 */
+	public File convert(Dump dump, String filePath) throws IOException {
+		File outputFile = createOutputFile(filePath);
+		if (outputFile == null) {
+			return null;
+		}
+		FileOutputStream output = new FileOutputStream(outputFile);
 		convertToReport(output, dump);
-		return output.toString(FileSystemUtils.UTF8_ENCODING);
+		return outputFile;
+	}
+
+	private File createOutputFile(String filePath) {
+		File outputFile = new File(filePath);
+
+		try {
+			boolean fileCreated = outputFile.createNewFile();
+			if (!fileCreated) {
+				return null;
+			}
+		} catch (IOException e) {
+			logger.error("Could not create file " + outputFile.getAbsolutePath() + ". ");
```

Comment:
please let the user know what that means and what they can do about it. Otherwise this error message is more confusing than helpful. They don't know what that file is supposed to do (ideally: they don't even know we cache stuff on disk).
Also: please **always** log the exception object. It almost always contains useful details and helps debug the root cause (e.g. disk full, permission problems, ...)
I'd prefer something like `Could not create temporary coverage file {file}. This is used to cache the coverage file on disk before uploading it to its final destination. This coverage is lost. Please fix the underlying issue to avoid losing coverage.`
or similar